### PR TITLE
Mimir 332 kontakter

### PR DESCRIPTION
**Pull request by erikgus**
This corresponds to the UI part of MIMIR-487 - Fetch contacts from statreg.
- Adds a **[CustomSelector](https://developer.enonic.com/guides/build-a-custom-selector)** to `article` **content-type** and `contacts` **part**. 
- the **custom-selector** uses the `service` defined in `services/contacts` to list options that can be chosen by the user
- Contacts can be added to an `article` (preferred) or directly to the `contact` part. 
   - Contacts added to an article will have higher precedence. (contacts in the part will be ignored if there are contacts found at the article level).

While rendering,
- The **custom-selector** returns a list of ids (corresponding to the contact that was selected by the user while designing the page)
- The renderer finds the corresponding contacts from the statreg node mounted at `no.ssb.statreg` / `contacts` and chunks it for display.

### CODE REVIEW

File: src/main/resources/site/parts/contact/contact.es6 [hunk 1]
```diff
@@ -25,11 +25,13 @@ exports.preview = (req) => renderPart(req)
 function renderPart(req) {
   const WIDTH = 3 // how many boxes in a row
   const part = getComponent() || req
+  const page = getContent()
   const contactIdList = []
 
-  part.config.contacts = part.config.contacts && util.data.forceArray(part.config.contacts) || []
+  //checks page content for contacts first, then part for contacts and creates array
+  const contactsList = page.data.contacts.contact ? util.data.forceArray(page.data.contacts.contact) : part.config.contacts ? util.data.forceArray(part.config.contacts) : []
```
Comment (prabu-ssb) [inline src/main/resources/site/parts/contact/contact.es6:32]: We could avoid nesting ternaries, which makes for difficult reading. How about a simpler --

```
const contentContacts = ...
const partContacts = ...

const contactsList = [
  ...contentContacts,
  ...partContacts
]
```

Comment (Garlov) [inline src/main/resources/site/parts/contact/contact.es6:32]: Error here when adding the part to a template, as well as linting warn/err

File: src/main/resources/site/content-types/article/article.xml [hunk 2]
```diff
@@ -180,5 +180,18 @@
                 </input>
             </items>
         </field-set>
+
+        <field-set>
+            <label>Kontakter</label>
+            <items>
+                <input name="contacts" type="CustomSelector">
+                    <occurrences minimum="1" maximum="0" />
```
Comment (Garlov) [inline src/main/resources/site/content-types/article/article.xml:188]: this shouldn't be forced, it should be possible to have an article without contacts. The part should then be hidden as well.

File: src/main/resources/site/parts/contact/contact.es6 [hunk 3]
```diff
@@ -1,12 +1,16 @@
...
 } = __non_webpack_require__( '/lib/xp/portal')
 const {
   render
 } = __non_webpack_require__( '/lib/thymeleaf')
 const {
   renderError
 } = __non_webpack_require__( '/lib/error/error')
+const { getContactsFromRepo } = __non_webpack_require__('/lib/repo/statreg/contacts')
+const { ensureArray } = __non_webpack_require__('/lib/polyfills/xp-util')
+const { chunkArray } = __non_webpack_require__('/lib/arrayUtils')
```
Comment (Garlov) [inline src/main/resources/site/parts/contact/contact.es6:12]: Should look into streamlining utils? We already have a /ssb/utils, and now we get two more. Is that really necessary?

**Comment by Garlov** [comment]
I can also add the 404 no contacts found contact: 
image

It should send back an empty array if there are no contacts, and status 500 if it fails to connect to the contacts repo or stat reg

File: src/main/resources/site/parts/contact/contact.es6 (hunk 3 above)
Comment (prabu-ssb) [reply src/main/resources/site/parts/contact/contact.es6:12]: I am for **streamlining** utils, but not putting them all in one `utils.es6` -- file just to keep them modular and only-related-utils-in-one-place. Agree that ensureArray and chunkArray might be good to be in one place - will club them together and move them to `/lib/ssb`

**Comment by prabu-ssb** [comment]
> I can also add the 404 no contacts found contact:
> image
> 
> It should send back an empty array if there are no contacts, and status 500 if it fails to connect to the contacts repo or stat reg

oops! this is a slip - should have been removed after local testing.. will fix! :)

**Comment by prabu-ssb** [comment]
> > I can also add the 404 no contacts found contact:
> > image
> > It should send back an empty array if there are no contacts, and status 500 if it fails to connect to the contacts repo or stat reg
> 
> oops! this is a slip - should have been removed after local testing.. will fix! :)

Observing something strange -- when I return 404 and an empty array, XP seems to report `Service Error: Unknown Error`

404-handling1
404-handling2

**Pull request merged**